### ISSUE TITLE
Steam client fails on Ubuntu 19.04

### ISSUE THREAD

**Issue by clrg** [davewx7/citadel#258]
Opening the game from Steam fails silently.

Apologies; don't know why code blocks are eating line breaks so you get sucky formatting instead.

Output from the terminal:

~/.steam/steam/steamapps/common/Argentum Age$ ./anura
./anura: error while loading shared libraries: libGLEW.so.1.10: cannot open shared object file: No such file or directory

~/.steam/steam/steamapps/common/Argentum Age$ find ~/.steam/ | grep -i libGLEW.so.1.10
/home/charles/.steam/ubuntu12_32/steam-runtime/i386/usr/lib/i386-linux-gnu/libGLEW.so.1.10
/home/charles/.steam/ubuntu12_32/steam-runtime/i386/usr/lib/i386-linux-gnu/libGLEW.so.1.10.0
/home/charles/.steam/ubuntu12_32/steam-runtime/amd64/usr/lib/x86_64-linux-gnu/libGLEW.so.1.10
/home/charles/.steam/ubuntu12_32/steam-runtime/amd64/usr/lib/x86_64-linux-gnu/libGLEW.so.1.10.0
/home/charles/.steam/steam/steamapps/common/Argentum Age/runtime/libGLEW.so.1.10

Running the sh:

~/.steam/steam/steamapps/common/Argentum Age$ ./anura.sh 
INFO: main.cpp:475 : Anura engine version 1.4
INFO: main.cpp:514 : Default Tile Size: 16
INFO: main.cpp:515 : Default Tile Scale: 2
INFO: main.cpp:517 : Build Options:
INFO: main.cpp:519 :     isomap
INFO: main.cpp:519 :     lua
INFO: main.cpp:519 :     save_png
INFO: main.cpp:519 :     sdl2
INFO: main.cpp:519 :     svg
INFO: module.cpp:334 : LOOKING IN 'modules/frogatto/module.cfg': 0
INFO: module.cpp:334 : LOOKING IN '/home/charles/.frogatto//dlc/frogatto/module.cfg': 0
INFO: main.cpp:593 : FAILED TO LOAD MODULE: frogatto

Oh, I'm notified on Discord that I need to specify '--module=citadel'...

~/.steam/steam/steamapps/common/Argentum Age$ ./anura.sh --module=citadel
INFO: main.cpp:475 : Anura engine version 1.4
INFO: main.cpp:514 : Default Tile Size: 16
INFO: main.cpp:515 : Default Tile Scale: 2
INFO: main.cpp:517 : Build Options:
INFO: main.cpp:519 :     isomap
INFO: main.cpp:519 :     lua
INFO: main.cpp:519 :     save_png
INFO: main.cpp:519 :     sdl2
INFO: main.cpp:519 :     svg
INFO: preferences.cpp:583 : SET PREFERENCES PATH: ~/.citadel/
INFO: module.cpp:334 : LOOKING IN 'modules/citadel/module.cfg': 1
INFO: preferences.cpp:583 : SET PREFERENCES PATH: ~/.citadel/
INFO: preferences.cpp:637 : EXPAND DATA PATHS
INFO: preferences.cpp:637 : EXPAND DATA PATHS
INFO: preferences.cpp:637 : EXPAND DATA PATHS
INFO: preferences.cpp:637 : EXPAND DATA PATHS
INFO: preferences.cpp:637 : EXPAND DATA PATHS
INFO: main.cpp:620 : ARGS: --module=citadel
INFO: main.cpp:620 : ARGS: --auto-size-window
INFO: main.cpp:620 : ARGS: --auto-size-ideal-width=1920
INFO: main.cpp:620 : ARGS: --auto-size-ideal-height=1080
INFO: main.cpp:620 : ARGS: --defer-archive-min-window-width=800
INFO: main.cpp:620 : ARGS: --defer-archive-min-window-height=600
INFO: main.cpp:620 : ARGS: --auto-anchor-objects
INFO: main.cpp:620 : ARGS: --set-fps=60
INFO: preferences.cpp:1082 : FPS: 60 = 16ms/frame
INFO: main.cpp:620 : ARGS: --no-autopause
INFO: main.cpp:620 : ARGS: --server=http://theargentlark.com:23456
INFO: main.cpp:620 : ARGS: --tbs-server
INFO: preferences.cpp:1121 : TURN ON internal server
INFO: main.cpp:620 : ARGS: --allow-debug-console-clicking=false
INFO: main.cpp:620 : ARGS: --disable-global-alpha-filter
INFO: main.cpp:620 : ARGS: --mouse-drag-threshold=100
INFO: main.cpp:620 : ARGS: --loading-screen-bg-color=#16232e
INFO: main.cpp:620 : ARGS: --reload-modified-objects
INFO: main.cpp:620 : ARGS: --start-rumble=false
INFO: main.cpp:620 : ARGS: --enable-graphical-fonts=false
INFO: main.cpp:620 : ARGS: --draw-objects-on-even-pixel-boundaries=false
INFO: main.cpp:620 : ARGS: --allow-builtin-settings-dialog=false
INFO: main.cpp:620 : ARGS: --disable-fullscreen
INFO: main.cpp:620 : ARGS: --max-frame-skips=2
INFO: main.cpp:620 : ARGS: --force-send-stats=false
INFO: main.cpp:620 : ARGS: --devtools=true
INFO: main.cpp:620 : ARGS: --particle-ui-show-save=false
INFO: main.cpp:620 : ARGS: --particle-ui-show-camera=false
INFO: main.cpp:620 : ARGS: --particle-ui-2d=true
INFO: main.cpp:620 : ARGS: --dump-to-console=false
INFO: main.cpp:620 : ARGS: --defer-archive-desktop-fullscreen=true
INFO: main.cpp:620 : ARGS: --log-instrumentation=true
INFO: main.cpp:620 : ARGS: --no-tests
INFO: main.cpp:620 : ARGS: --ffl-vm=true
INFO: preferences.cpp:637 : EXPAND DATA PATHS
INFO: main.cpp:754 : Preferences dir: /home/charles/.citadel/
dbus[7105]: arguments to dbus_type_is_basic() were incorrect, assertion "dbus_type_is_valid (typecode) || typecode == DBUS_TYPE_INVALID" failed in file ../../../dbus/dbus-signature.c line 323.
This is normally a bug in some application using the D-Bus library.

  D-Bus not built with -rdynamic so unable to print a backtrace
./anura.sh: line 3:  7105 Aborted                 (core dumped) LD_LIBRARY_PATH=./runtime:$LD_LIBRARY_PATH ./anura "$@"

Eep!

**Comment by uprego** [comment]
Related: ein_waschechter_yugioh_lappen#4267@Discord
[claims that the Steam distribution "runs just fine"][0]
in her/his [ubuntu 19.04 installation][1].

[0]: https://discordapp.com/channels/156241487938584577/447492980098596876/600714781367861256

[1]: https://discordapp.com/channels/156241487938584577/447492980098596876/600711341464092673

**Comment by galegosimpatico** [comment]
Would suggest labelling [irreproducible][0] **but** leaving open issue for now.


[0]: https://github.com/davewx7/citadel/labels/irreproducible

**Comment by arthurlutz** [comment]
I think the steam build depends on libglew 1.10 

You can "start" to solve this by installing the old libglew1.10 by finding the package libglew1.10_1.10.0-3_amd64.deb This is probably not recommended but might solve the issue. You might then get another dependency error (stx something if I remember correctly).

**Comment by GSchuemann** [comment]
Hi, I am the person who claimed it ran on Ubuntu 19.04 but having the exact same issue on 20.04 now when using the Steam Build. Running Anura with: LD_LIBRARY_PATH="/home/$USER/.steam/steam/steamapps/common/Argentum Age/runtime" ./anura --module=citadel

libglew 1.10 is in the library Path and maybe the build should be updated ? I mean libglew1.10 is quite a few years old now.

**Comment by uprego** [comment]
> [...]
> 
> libglew 1.10 is in the library Path and maybe the build should be updated ? I mean libglew1.10 is quite a few years old now.

You could expect it to get raised up to date after some proof that it's is so aged it causes issues. I doubt it any time before.

You could develop that update yourself, then I guess you should make sure no Windows, Linux or macOS builds suffer. Related, there's no longer a working CI configuration for Anura ([no surprise it's broken at the time of writing][0]).

[0]: https://github.com/anura-engine/anura/issues/303

**Comment by galegosimpatico** [comment]
Dbus type is invalid, and exporting a new dbus launch (`export $(dbus-launch)`) doesn't make this go away, right?

**Comment by GSchuemann** [comment]
No, but got a bit further by simply googling the error. I found this: https://steamcommunity.com/app/561740/discussions/0/1753528725227441171/ for the game "MidBoss". So I tried LD_LIBRARY_PATH="/home/$USER/.steam/steam/steamapps/common/Argentum Age/runtime" LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libSDL2-2.0.so.0 ./anura --module=citadel

and I get past the dbus error and got now:
./anura: symbol lookup error: /lib/x86_64-linux-gnu/libxcb-dri3.so.0: undefined symbol: xcb_send_request_with_fds

Adding a LIBGL_DRI3_DISABLE=1 to the launch commands let's me start the game.

**Comment by galegosimpatico** [comment]
Awesome!

I think I've read some of those scattered at the Discord server, but you made an extraordinary compendia! :)
